prowler/providers/aws/services/cloudwatch/cloudwatch_log_group_no_critical_pii_in_logs/cloudwatch_log_group_no_critical_pii_in_logs.py

```diff
@@ -0,0 +1,137 @@
+                                        < cumulative_char_count + log_event_length
+                                    ):
+                                        flagged_event = log_group.log_streams[
+                                            log_stream_name
+                                        ][j]
+                                        cloudwatch_timestamp = (
+                                            convert_to_cloudwatch_timestamp_format(
+                                                flagged_event["timestamp"]
+                                            )
+                                        )
+                                        if cloudwatch_timestamp not in log_stream_pii:
+                                            log_stream_pii[cloudwatch_timestamp] = (
+                                                SecretsDict()
+                                            )
+
+                                        # Add the detected PII entity to log_stream_pii
+                                        log_stream_pii[cloudwatch_timestamp].add_secret(
+                                            pii.start - cumulative_char_count,
+                                            pii.entity_type,
+                                        )
+                                cumulative_char_count += (
+                                    log_event_length + 1
+                                )  # +1 to account for '\n'
+
+                        if log_stream_pii:
+                            pii_string = "; ".join(
+                                [
+                                    f"at {timestamp} - {log_stream_pii[timestamp].to_string()}"
+                                    for timestamp in log_stream_pii
+                                ]
+                            )
+                            log_group_pii.append(
+                                f"in log stream {log_stream_name} {pii_string}"
+                            )
+                if log_group_pii:
+                    pii_string = "; ".join(log_group_pii)
+                    report.status = "FAIL"
+                    report.status_extended = f"Potential critical PII found in log group {log_group.name} {pii_string}."
+                findings.append(report)
+        return findings
+
+
+class SecretsDict(dict):
+    # Using this dict to remove duplicates of the PII type showing up multiple times on the same line
+    # Also includes the to_string method
+    def add_secret(self, line_number, pii_type):
+        if line_number not in self.keys():
+            self[line_number] = [pii_type]
+        else:
+            if pii_type not in self[line_number]:
+                self[line_number] += [pii_type]
+
+    def to_string(self):
+        return ", ".join(
+            [
+                f"{', '.join(pii_types)} on line {line_number}"
+                for line_number, pii_types in sorted(self.items())
+            ]
+        )
```

Comment:
```suggestion
class SecretsDict(dict[int, Set[str]]):
    """Dictionary to track unique PII types on each line."""

    def add_secret(self, line_number: int, pii_type: str) -> None:
        """Add a PII type to a specific line number, ensuring no duplicates."""
        self.setdefault(line_number, set()).add(pii_type)

    def __str__(self) -> str:
        """Generate a formatted string representation of the dictionary."""
        return ", ".join(
            f"{', '.join(sorted(pii_types))} on line {line_number}"
            for line_number, pii_types in sorted(self.items())
        )
```
Considering that you does not want duplicates in PII types, I think a set is more suitable data structure.